components/raftstore/src/store/fsm/peer.rs

```diff
@@ -6555,6 +6557,66 @@ where
     fn register_report_region_buckets_tick(&mut self) {
         self.schedule_tick(PeerTick::ReportBuckets)
     }
+
+    fn register_check_peer_complete_recovery(&mut self) {
+        self.schedule_tick(PeerTick::CheckPeerCompleteRecovery)
+    }
+
+    fn on_check_peer_complete_recovery(&mut self) {
+        if !self.fsm.peer.pending_recovery {
+            return;
+        }
+
+        // If the peer is newly added or created, no need to check the recovery status.
+        let peer_id = self.fsm.peer.peer_id();
+        let is_newly_added_peer = self
+            .fsm
+            .peer
+            .raft_group
+            .raft
+            .prs()
+            .get(peer_id)
+            .map_or(false, |pr| pr.matched == 0);
+        if is_newly_added_peer {
```

Comment:
Should we skip leader too?

Reply:
Yep, as this tick is only focus on the stage of `starting` progress of the store, if the peer is newly added (by split), whether its `leader` or `follower`, it's no need to tracking its state of applying.